src/hotspot/share/opto/callnode.cpp

```diff
@@ -1465,17 +1468,21 @@ SafePointScalarObjectNode::SafePointScalarObjectNode(const TypeOopPtr* tp,
                                                      Node* alloc,
 #endif
                                                      uint first_index,
-                                                     uint n_fields) :
+                                                     uint n_fields,
+                                                     int merge_pointer_idx,
+                                                     uint number_of_objects) :
   TypeNode(tp, 1), // 1 control input -- seems required.  Get from root.
   _first_index(first_index),
-  _n_fields(n_fields)
+  _n_fields(n_fields),
 #ifdef ASSERT
-  , _alloc(alloc)
+  _alloc(alloc),
 #endif
```

Comment:
May be we should always pass alloc, even in product VM. It is not related to your changes but it is pain to have.

Reply:
I can make that change.